src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMNavigator.java

```diff
@@ -210,6 +225,81 @@ private Object readResolve() throws ObjectStreamException {
         return this;
     }
 
+    /*
+     * Optionally update bitbucketJenkinsRootUrl if it was not set
+     * before (which causes normalization of the URL string value),
+     * or if the cached endpointcfgJenkinsRootUrl changed. Also updates
+     * this cached value as needed (was not set or did change).
+     */
+    @SuppressFBWarnings(value = "RCN_REDUNDANT_NULLCHECK_OF_NONNULL_VALUE",
```

Comment:
This seems like code smell

Reply:
At least in the current iteration (or some earlier points in the development intent), there is a difference between a `null` value which means that no setting was applied into the string yet (so it should be re-requested from the endpoint configs) vs. a known empty-string setting (so we have loaded a config, and it did not specify a custom URL).

Either way, when we do not have a definitive custom URL set, we return a (normalized) global Jenkins Root URL. Normalization of the webhook URL compared to what users can enter proved important, e.g. a missing trailing slash causes mayhem. Maybe ensuring of this slash can be moved to where the value is used, but then the code would be duplicated there, and I was not sure I tracked down all the consumers :\